pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -484,41 +486,53 @@ func (c *Controller) addNodeRoute(nodeName string, node *v1.Node) error {
-// tunnel does not exist.
-func (c *Controller) createIPSecTunnelPort(nodeName string, nodeIP net.IP) error {
+// tunnel does not exist, and returns the ofport number.
+func (c *Controller) createIPSecTunnelPort(nodeName string, nodeIP net.IP) (int32, error) {
 	interfaceConfig, ok := c.interfaceStore.GetNodeTunnelInterface(nodeName)
 	if ok {
 		// TODO: check if Node IP, PSK, or tunnel type changes. This can
 		// happen if removeStaleTunnelPorts fails to remove a "stale"
 		// tunnel port for which the configuration has changed.
-		return nil
+		if interfaceConfig.OFPort != 0 {
+			return interfaceConfig.OFPort, nil
+		}
+	} else {
+		portName := util.GenerateNodeTunnelInterfaceName(nodeName)
+		ovsExternalIDs := map[string]interface{}{ovsExternalIDNodeName: nodeName}
+		portUUID, err := c.ovsBridgeClient.CreateTunnelPortExt(
+			portName,
+			c.networkConfig.TunnelType,
+			0, // ofPortRequest - let OVS allocate OFPort number.
+			nodeIP.String(),
+			c.networkConfig.IPSecPSK,
+			ovsExternalIDs)
+		if err != nil {
+			klog.Errorf("Failed to create OVS IPSec tunnel port for Node %s: %v", nodeName, err)
+			return 0, fmt.Errorf("failed to create IPSec tunnel port for Node %s", nodeName)
+		}
+		klog.Infof("Created IPSec tunnel port %s for Node %s", portName, nodeName)
+
+		ovsPortConfig := &interfacestore.OVSPortConfig{PortUUID: portUUID}
+		interfaceConfig = interfacestore.NewIPSecTunnelInterface(
+			portName,
+			c.networkConfig.TunnelType,
+			nodeName,
+			nodeIP,
+			c.networkConfig.IPSecPSK)
+		interfaceConfig.OVSPortConfig = ovsPortConfig
+		c.interfaceStore.AddInterface(interfaceConfig)
 	}
 
-	portName := util.GenerateNodeTunnelInterfaceName(nodeName)
-	ovsExternalIDs := map[string]interface{}{ovsExternalIDNodeName: nodeName}
-	portUUID, err := c.ovsBridgeClient.CreateTunnelPortExt(
-		portName,
-		c.networkConfig.TunnelType,
-		0, // ofPortRequest - let OVS allocate OFPort number.
-		nodeIP.String(),
-		c.networkConfig.IPSecPSK,
-		ovsExternalIDs)
+	// GetOFPort will wait for up to 1 second for OVSDB to report the OFPort number.
+	ofPort, err := c.ovsBridgeClient.GetOFPort(interfaceConfig.InterfaceName)
 	if err != nil {
-		klog.Errorf("Failed to create OVS IPSec tunnel port for Node %s: %v", nodeName, err)
-		return fmt.Errorf("failed to create IPSec tunnel port for Node %s", nodeName)
-	}
-	klog.Infof("Created IPSec tunnel port %s for Node %s", portName, nodeName)
-
-	ovsPortConfig := &interfacestore.OVSPortConfig{PortUUID: portUUID}
-	interfaceConfig = interfacestore.NewIPSecTunnelInterface(
-		portName,
-		c.networkConfig.TunnelType,
-		nodeName,
-		nodeIP,
-		c.networkConfig.IPSecPSK)
-	interfaceConfig.OVSPortConfig = ovsPortConfig
-	c.interfaceStore.AddInterface(interfaceConfig)
-	return nil
+		// Could be a temporary OVSDB connection failure or timeout.
+		// Let NodeRouteController retry at errors.
+		klog.Errorf("Failed to get of_port of the tunnel port for Node %s: %v", nodeName, err)
```

Comment:
same comment as above, I think we will end up logging the same error twice

Reply:
Sure removed the log.